src/coreclr/jit/morph.cpp

```diff
@@ -12478,6 +12484,65 @@ GenTree* Compiler::fgOptimizeEqualityComparisonWithConst(GenTreeOp* cmp)
             op1->SetVNsFromNode(cmp);
 
             DEBUG_DESTROY_NODE(cmp);
+
+            if (op1->OperIs(GT_GE, GT_LE) && (op1->gtGetOp2()->IsIntegralConst() || op1->gtGetOp1()->IsIntegralConst()))
```

Comment:
The idea here is that GT_EQ - GT_LT/GT_GT is reversed by the code above and reduced to GT_GE/GT_LE.